File: frontend/features/dashboard/schemas.ts
export type CategoryBreakdown = {
  category_id: string
  name: string
  color: string
  total: string
}

export type ExpenseGroupBreakdown = {
  total: string
  categories: CategoryBreakdown[]
}

export type DashboardSummaryData = {
  month_key: string
  card: ExpenseGroupBreakdown
  other: ExpenseGroupBreakdown
  month_total: string
  monthly_net_salary: string | null
  remaining: string | null
  percent_used: string | null
}

export type MonthlyOverview = {
  month_key: string
  month: number
  card_total: string
  other_total: string
  month_total: string
}

export type YearlyOverviewData = {
  year: number
  months: MonthlyOverview[]
  year_total: string
}

File: services/api/app/features/dashboard/schemas.py
from decimal import Decimal

from pydantic import BaseModel


class CategoryBreakdown(BaseModel):
    category_id: str
    name: str
    color: str
    total: Decimal


class ExpenseGroupBreakdown(BaseModel):
    total: Decimal
    categories: list[CategoryBreakdown]


class DashboardSummaryResponse(BaseModel):
    month_key: str
    card: ExpenseGroupBreakdown
    other: ExpenseGroupBreakdown
    month_total: Decimal
    monthly_net_salary: Decimal | None
    remaining: Decimal | None
    percent_used: Decimal | None
